src/main/java/com/redhat/pantheon/servlet/module/ReleaseDraftVersion.java

```diff
@@ -51,6 +51,16 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
         if( !versionToRelease.isPresent() ) {
             response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
                     "The module doesn't have a draft version to be released");
+        } else if (versionToRelease.get().metadata.getOrCreate().productVersion.get() == null
+        		||  versionToRelease.get().metadata.getOrCreate().productVersion.get().isEmpty()) {
+        	// Check if productVersion is set
+        	response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
+                    "The version to be released doesn't have productVersion metadata");
+        } else if (versionToRelease.get().metadata.getOrCreate().urlFragment.get() == null
+        		||  versionToRelease.get().metadata.getOrCreate().urlFragment.get().isEmpty()) {
+        	// Check if urlFragment is set
+        	response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
+                    "The version to be released doesn't have urlFragment metadata");
```

Comment:
Tested and it works!

Because I'm a pain-in-the-ass perfectionist, I have to point out that technically, this code doesn't match what the UI checks for. The UI only checks for productVersion, but this checks for both productVersion and urlFragment.

That kind of works because the UI is operating on an all-or-nothing basis right now, but that might not always be the case. The cleanest implementation of this would check for everything that's relevant (product version, url fragment, and search keywords) in both places, but since we've gone back and forth on this PR enough already, let's save those details for another day.